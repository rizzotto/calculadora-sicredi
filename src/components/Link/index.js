import React from "react";
import { Link as RRLink } from "react-router-dom";

import { makeStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";

import theme from "../../styles/customMuiTheme.js";

const useStyles = makeStyles({
    link: {
        color: theme.palette.primary.main,
        fontSize: 18,
        fontWeight: 800,
        textDecoration: "none",
    },
    white: {
        color: "#FFF",
        fontSize: 18,
        fontWeight: 800,
        textDecoration: "none",
        backgroundColor: theme.palette.quaternary.main,
        width: "100%",
        maxWidth: 500,
        height: 56,
        borderRadius: 8,
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        "&:hover": {
            backgroundColor: theme.palette.quaternary.main,
            opacity: 0.9,
        },
    },
});

export default function Link({ children, to, color }) {
    const classes = useStyles();

    return (
        <RRLink
            to={to}
            className={color === "white" ? classes.white : classes.link}
        >
            {children}
        </RRLink>
    );
}
Link.propTypes = {
    children: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    color: PropTypes.string,
};
