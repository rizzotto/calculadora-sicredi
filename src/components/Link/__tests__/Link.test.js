import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import { Typography } from "@material-ui/core";

import Link from "../index.js";

it("renders link correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Link to={"/test"}>
                    <Typography>{"teste"}</Typography>
                </Link>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
