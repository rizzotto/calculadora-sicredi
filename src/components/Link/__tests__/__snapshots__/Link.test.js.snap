// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link correctly 1`] = `
<a
  className="makeStyles-link-1"
  href="/test"
  onClick={[Function]}
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    teste
  </p>
</a>
`;
